vhost/src/vhost_user/backend_req_handler.rs

```diff
@@ -555,6 +561,24 @@ impl<S: VhostUserBackendReqHandler> BackendReqHandler<S> {
                 let res = self.set_backend_req_fd(files);
                 self.send_ack_message(&hdr, res)?;
             }
+            Ok(FrontendReq::GET_SHARED_OBJECT) => {
+                self.check_proto_feature(VhostUserProtocolFeatures::SHARED_OBJECT)?;
+                self.check_request_size(&hdr, size, 0)?;
+                let msg = self.extract_request_body::<VhostUserSharedMsg>(&hdr, size, &buf)?;
+                let res = self.backend.get_shared_object(msg);
+
+                match res {
+                    Ok(file) => {
+                        let hdr = self.new_reply_header::<VhostUserSharedMsg>(&hdr, 0)?;
+                        self.main_sock
+                            .send_message(&hdr, &msg, Some(&[file.as_raw_fd()]))?
+                    }
+                    Err(_) => {
+                        let msg = VhostUserSharedMsg::default();
+                        self.main_sock.send_message(&hdr, &msg, None)?;
```

Comment:
Why are we re-using the `hdr` of the request?
BTW also in this case I'm not sure if it's a good idea using `VhostUserSharedMsg`, so maybe just call   `self.send_ack_message(&hdr, res)?;` or send an empty message without any fd should be fine, the spec are not really clear.

Reply:
yes, sending an empty message is fine, also I will remove the `hdr` and `msg` of `VhostUserSharedMsg`.  Thanks :) 

> the spec are not really clear.

+1

Reply:
Frontend implementation basically only checks for request type and the fd for the replies, so backends can do anything as long as they leave the ancillary fds empty. 

I think empty message is fine.

On the other hand, I think this is the first time someone looked into an actual implementation of the backend side to support this message type. If you think it would be more useful to define the return type otherwise, we could propose a change to the spec.

Reply:
IIUC most commands return a `u64` with some kind of return value, zero on success or non-zero on failure. There is also the `VHOST_USER_PROTOCOL_F_REPLY_ACK` section talking about that. So maybe we should follow that, but IMHO that should be more clear in the spec in general, not for this specific command.

Reply:
The issue with `hdr` is still there. Can you define a `reply_hdr` just before the match, to be sure that you're not reusing the incoming `hdr` ?